sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -111,17 +117,19 @@ def test_compute_mi_cd_unique_label():
 
 
 # We are going test that feature ordering by MI matches our expectations.
-def test_mutual_info_classif_discrete():
-    X = np.array([[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]])
-    y = np.array([0, 1, 2, 2, 1])
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_mutual_info_classif_discrete(dtype):
+    X = np.array([[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]], dtype=dtype)
+    y = np.array([0, 1, 2, 2, 1], dtype=dtype)
```

Comment:
Since this is a classification target, it does not make sense to convert y as floating point values.